server/src/main/java/org/apache/seata/server/cluster/raft/manager/RaftControllerStateMachine.java

```diff
@@ -0,0 +1,738 @@
+        for (String groupName : groupNames) {
+            if (StringUtils.isBlank(groupName.trim())) {
+                LOGGER.error("TXG group name cannot be empty. Groups: {}", Arrays.toString(groupNames));
+                return false;
+            }
+        }
+
+        LOGGER.info("TXG requirements validation passed: {} groups, {} nodes",
+                groupNames.length, controllerMembers.size());
+        return true;
+    }
+
+    private Map<String, List<String>> createTxgGroupAssignments(String[] groupNames, List<PeerId> members) {
+        Map<String, List<String>> assignments = new HashMap<>();
+
+        // Convert PeerId list to string list (IP:Port format)
+        List<String> memberEndpoints = members.stream()
+                .map(peerId -> peerId.getIp() + ":" + peerId.getPort())
+                .collect(Collectors.toList());
+
+        //since there are only 3 nodes, assign all nodes to all groups
+        // will need a better algorithm if we need to have dynamic numbers but for now lets keep it simple?
+        // TXG-1: 192.168.1.100,192.168.1.101,192.168.1.102
+        // TXG-2: 192.168.1.100,192.168.1.101,192.168.1.102
+        // TXG-3: 192.168.1.100,192.168.1.101,192.168.1.102
+        for (String groupName : groupNames) {
+            String trimmedGroupName = groupName.trim();
+            assignments.put(trimmedGroupName, new ArrayList<>(memberEndpoints));
+            LOGGER.info("Assigned all {} nodes to group {}: {}",
+                    memberEndpoints.size(), trimmedGroupName, memberEndpoints);
+        }
+
+        return assignments;
+    }
+
+    private void submitTxgGroupAssignments(Map<String, List<String>> assignments) {
+        try {
+            // Create the DTO
+            TxgGroupAssignmentDTO txgAssignments = new TxgGroupAssignmentDTO(assignments);
+
+            // Create the message
+            RaftTxgGroupMsg message = new RaftTxgGroupMsg(txgAssignments);
+            message.setGroup(group); // Set the controller group
+
+            // Submit to state machine so all controller followers get the assignments
+            RaftTaskUtil.createTask(
+                    status -> {
+                        if (status.isOk()) {
+                            LOGGER.info("TXG group assignments successfully replicated to all controller nodes");
+                            // After successful replication, do we need to trigger something?
+                            triggerTxgConstruction();
+                        } else {
+                            LOGGER.error("Failed to replicate TXG assignments: {}", status.getErrorMsg());
+                        }
+                    },
+                    message,  // The message containing group assignments
+                    null
+            );
+
+        } catch (Exception e) {
+            LOGGER.error("Failed to submit TXG group assignments", e);
+        }
+    }
+
+    private void triggerTxgConstruction() {
+        //spin-wait here?
```

Comment:
Not here. This is the CG startup process, and CG does not require spin-waiting. You need to store the allocation information of TXG somewhere, such as in RaftControllerServerManager, or define a separate class specifically for storing TXG metadata. Additionally, the follower also needs to define a corresponding execute method to process the contents of the passed message, updating or adding TXG information accordingly.